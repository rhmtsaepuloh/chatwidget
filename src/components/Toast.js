import React from "react";

const Toast = ({ text, display }) => {
  return (
    <div
      className='footer-chatroom absolute mx-auto items-center justify-center'
      style={{
        bottom: "100px",
        display: display ? "flex" : "none",
        zIndex: 2,
      }}>
      <div className='bg-red-400 rounded-full py-2 px-8 '>
        <p className='text-center text-white'>{text}</p>
      </div>
    </div>
  );
};

export default Toast;
