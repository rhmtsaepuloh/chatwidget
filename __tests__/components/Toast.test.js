/* eslint-disable react/react-in-jsx-scope */
import React from "react";
import { shallow } from "enzyme";
import Toast from "../../src/components/Toast";

const text = "Chat Aja";

describe("render correctly toast components", () => {
  test("renders if conditional display true", () => {
    const display = true;
    const wrapper = shallow(<Toast text={text} display={display} />);
    expect(wrapper).toMatchSnapshot();
  });
  test("renders if conditional display false", () => {
    const display = false;
    const wrapper = shallow(<Toast text={text} display={display} />);
    expect(wrapper).toMatchSnapshot();
  });
});
