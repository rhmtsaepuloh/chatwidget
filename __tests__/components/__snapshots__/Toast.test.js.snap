// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly toast components renders if conditional display false 1`] = `ShallowWrapper {}`;

exports[`render correctly toast components renders if conditional display true 1`] = `ShallowWrapper {}`;
